godot-core/src/builtin/collections/packed_array.rs

```diff
@@ -433,10 +452,38 @@ macro_rules! impl_packed_array {
             }
         }
 
+        #[doc = concat!("Creates a `", stringify!($PackedArray), "` from the given Rust array.")]
+        impl<const N: usize> From<[$Element; N]> for $PackedArray {
+            fn from(arr: [$Element; N]) -> Self {
+                let mut ret = match N {
+                    0 => return Self::new(),
+                    len => Self::default_with_size(len),
+                };
+
+                // SAFETY: The packed array contains exactly N elements and the source array will be forgotten.
+                unsafe {
+                    let arr = std::mem::ManuallyDrop::new(arr);
+                    ret.move_from_slice(arr.as_ptr(), N);
+                }
+                ret
+            }
+        }
+
         #[doc = concat!("Creates a `", stringify!($PackedArray), "` from the given Rust vec.")]
         impl From<Vec<$Element>> for $PackedArray {
-            fn from(vec: Vec<$Element>) -> Self{
-                vec.into_iter().collect()
+            fn from(mut vec: Vec<$Element>) -> Self {
+                let (len, mut ret) = match vec.len() {
+                    0 => return Self::new(),
+                    len => (len, Self::default_with_size(len)),
+                };
```

Comment:
This is unnecessarily hard to understand, could you just write a
```rs
if vec.is_empty() {
    return Self::new();
}
```
instead?

Reply:
So something like this?

```rust
if vec.is_empty() {
    return Self::new();
}
let len = vec.len();
let mut ret = Self::default_with_size(len);
```

Reply:
Yeah, maybe I was too optimistic with the code reuse here... Otherwise it's OK to not have `default_with_size`. More important than a bit of duplication or fewer lines of code is readability (which is problematic with the `match` version).